File: Python/0080. 密码发生器.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            n = int(input())
            for _ in range(n):
                s = input()
                res = [0] * 6
                for i in range(len(s)):
                    res[i % 6] += ord(s[i])
                for x in res:
                    temp = x
                    tt = 0
                    while temp >= 10 :
                        while temp:
                            tt += temp % 10
                            temp //= 10
                        temp = tt
                        tt = 0
                    print(temp, end = '')
                print('')                      
        
        except EOFError:
            break
    
if __name__ == "__main__":
    func()

File: C++/0080. 密码发生器.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.
#include <vector>
int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    int n;
    cin>>n;
    string s;
    while(n > 0) {
        n--;
        cin>>s;
        vector<int> res (6, 0);
        for(int i=0;i<s.size();++i) {
            res[i % 6] += int(s[i]);
        }
        for(auto x:res) {
            int temp = x;
            int tt = 0;
            while(temp >= 10) {
                while(temp != 0) {
                    tt += temp % 10;
                    temp /= 10;
                }
                temp = tt;
                tt = 0;       
            }
            cout<<temp;
        }
        cout<<endl;
    }
    return 0;
}
